### PR TITLE
Added Option.onEmpty() (#1781)

### PR DESCRIPTION
**Pull request by Opalo**
Hi,

As discussed in #1781 I've provided `onEmpty` method to perform side effects for `Option` class.

### CODE REVIEW

File: javaslang/src/main/java/javaslang/control/Option.java [hunk 1]
```diff
@@ -32,6 +32,7 @@
  *
  * @param <T> The type of the optional value.
  * @author Daniel Dietrich
+ * @author Maciek Opala
```
Comment (ruslansennov) [inline javaslang/src/main/java/javaslang/control/Option.java]: I believe this is too small contribution to add @author annotation

Comment (Opalo) [inline javaslang/src/main/java/javaslang/control/Option.java]: @ruslansennov sure! Wasn't sure about contribution rules. Will remove it.

File: javaslang/src/main/java/javaslang/control/Option.java [hunk 2]
```diff
@@ -171,6 +171,19 @@
     boolean isEmpty();
 
     /**
+     * Runs a Java Runnable passed as parameter if this {@code Option} is empty.
+     *
+     * @param action a given Runnable to be run
+     * @return this {@code Option}
+     */
+    default Option<T> onEmpty(Runnable action) {
```
Comment (danieldietrich) [inline javaslang/src/main/java/javaslang/control/Option.java]: Please add a null-check, we do it by convention for better readability of stack traces:

```java
Objects.requireNonNull(action, "action is null");
```

Comment (Opalo) [inline javaslang/src/main/java/javaslang/control/Option.java]: @danieldietrich you are absolutely right - sorry for not adding it. I've amended the commit with required null-check.

Comment (danieldietrich) [inline javaslang/src/main/java/javaslang/control/Option.java]: Great, thank you!

**Comment by danieldietrich** [comment]
Lovely feature, thank you for your patience 🙏

**Comment by Opalo** [comment]
Always happy to help! Thanks for accepting the PR, we really need this feature :)

**Comment by danieldietrich** [comment]
I will ship 2.1.0-beta as soon as possible. However, it will still take some time...

**Pull request merged**